nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,328 @@
+                        future.complete(statuses1);
+                    }
+                } catch (Exception e) {
+                    future.completeExceptionally(new ProcessException("Error occurred while checking ingestion status", e));
+                }
+            };
+            scheduler.scheduleWithFixedDelay(task, 1, 2, TimeUnit.SECONDS);
+            statuses = future.get(1800, TimeUnit.SECONDS);
+            return KustoIngestionResult.fromString(statuses.get(0).status.toString());
+        } catch (IngestionClientException | IngestionServiceException e) {
+            throw new ProcessException("Error occurred while ingesting data into ADX", e);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw new ProcessException("Error occurred while checking ingestion status", e);
+        } finally {
+            shutDownScheduler(scheduler);
+        }
+    }
+
+    public void shutDownScheduler(ScheduledExecutorService executorService) {
+        executorService.shutdown();
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                // Cancel currently executing tasks forcefully
+                executorService.shutdownNow();
+                // Wait a while for tasks to respond to being cancelled
+                if (!executorService.awaitTermination(60, TimeUnit.SECONDS))
+                    getLogger().error("Scheduler did not terminate");
+            }
+        } catch (InterruptedException ex) {
+            // (Re-)Cancel if current thread also interrupted
+            executorService.shutdownNow();
+        }
+    }
+
+    private ConnectionStringBuilder createKustoEngineConnectionString(final String clusterUrl,
+                                                                      final String appId,
+                                                                      final String appKey,
+                                                                      final String appTenant,
+                                                                      final KustoAuthenticationStrategy kustoAuthStrategy) {
+        final ConnectionStringBuilder builder;
+        if(KustoAuthenticationStrategy.APPLICATION_CREDENTIALS == kustoAuthStrategy){
+            builder = ConnectionStringBuilder.createWithAadApplicationCredentials(clusterUrl, appId, appKey, appTenant);
+        }else{
+            builder = ConnectionStringBuilder.createWithAadManagedIdentity(clusterUrl, appId);
+        }
+
+        final String vendor = System.getProperty("java.vendor");
+        final String version = System.getProperty("java.version");
+
+        builder.setConnectorDetails("Kusto.Nifi", StandardKustoIngestService.class.getPackage().getImplementationVersion(), vendor, version, false, null, NIFI_SINK);
```

Comment:
Should the vendor and version be another pair ? 

Also AppId/version are not the JDK versions (https://github.com/Azure/azure-kusto-java/blob/eff584c119a171f5f046dc66a3bb94881eba991d/data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java#L379) They are probably intended for NiFi versions ? e.g. Connector version 1.0.0 and Nifi version 2.0.0. If that is too hard, null it

They are for appId and version ( get defaulted to SDK versions - https://github.com/Azure/azure-kusto-java/blob/eff584c119a171f5f046dc66a3bb94881eba991d/data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java#L93)

Reply:
agreed Ram. initial commit was with appname and appversion as null but it was added by the nifi folks in ADX source PR, I tried to accommodate that change by adding the java version details to the appName and appVersion.